text/0013-portable-pdb.md

```diff
@@ -0,0 +1,86 @@
+- Start Date: 2022-09-22
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/13
+
+# Summary
+
+This RFC proposes new protocol fields to help with symbolicating .NET stack traces that use Portable PDBs internally for
+symbolication.
+
+# Motivation
+
+To symbolicate a stack frame using a Portable PDB, we need the following data:
+
+- IL Offset
+- Method Index
+
+Additionally, fetching a Portable PDB from an external Symbol server following the [SSQP specification](https://github.com/dotnet/symstore/blob/main/docs/specs/SSQP_Key_Conventions.md#portable-pdb-signature) needs to have:
+
+- PDB File name
+- PDB Id
+- PDB Checksum
+
+Currently the `SentryStackFrame` type of the `sentry-dotnet` SDK has an
+[`instruction_offset` field](https://github.com/getsentry/sentry-dotnet/blob/57044ff52320c82cf1ca22cceee7125dfb9d1423/src/Sentry/SentryStackFrame.cs#L128-L135),
+which is itself not yet documented on [`develop.sentry.dev`](https://develop.sentry.dev/sdk/event-payloads/stacktrace/#frame-attributes).
+
+# Background
+
+The reason this decision or document is required. This section might not always exist.
+
+# Options Considered
+
+## Stack Trace Interface Extensions
+
+We propose to add the following to the [Stack Trace Interface Frame Attributes](https://develop.sentry.dev/sdk/event-payloads/stacktrace/#frame-attributes):
+
+- `instruction_offset: Number`, as it is currently used by the `sentry-dotnet` SDK.
+- `method_index: Number`.
+
+The `instruction_offset` is defined as a JSON `Number`.
```

Comment:
https://github.com/dotnet/runtime/blob/main/docs/design/specs/PortablePdb-Metadata.md#sequence-points-blob
This defines that:

> IL Offset is within range [0, 0x20000000)

I believe the ECMA 335 spec also has tight limits on what the IL offset can be.
This should be well in the range what JS/ON `Number` can safely hold.

If we want to change this, this would also need a breaking change in the `sentry-dotnet` SDK. CC @mattjohnsonpint

Reply:
We currently send the `instruction_addr` in hex format and the `instruction_offset` as a long integer.  We could easily change both to be sent in hex format without requiring a "breaking" change - assuming the back end will accept either format.

Reply:
I think even if it's unlikely that this values goes out of the currently defined 32bit range, better safe than sorry and implement as hex field. SDKs can keep sending integers if they are capable of fitting it into their JSON int.

Reply:
Looks like `instruction_offset` isn't retained anymore, after https://github.com/getsentry/sentry/pull/4892?

Reply:
Oh, that is interesting, looks like `instruction_offset` meant something entirely different previously: The offset from the start of the symbol? (just going by the description of the linked PR)

Reply:
The protocol generally accepts both `0x`-prefixed hex and numeric representations for address-like fields. As such, you do not have to consider it a breaking change if you switch the SDK from one to another representation. 

I agree with @mitsuhiko's ask to make this an `Addr` type to allow for large values. The SDK may still send numbers as long as it can guarantee that they do not exceed 53 bit significance.